gui/gui.go

```diff
@@ -514,9 +509,11 @@ Buffer Size: %d lines
 // redrawn. Limiting is applied on wakeups to avoid excessive CPU
 // usage when the terminal is being updated rapidly.
 func (gui *GUI) waker(stop <-chan struct{}) {
+	gui.wakerEnded = make(chan bool)
```

Comment:
with a WaitGroup, the only change required here would a single line like: `defer gui.wg.Done()`